### PR TITLE
Issue-635 ApiExtensions package

### PR DESCRIPTION
**Pull request by worryg0d**
This PR adds `apiextensions` package which provides shared objects as:
* `ObjectReference`
* `ObjectFieldReference`

### CODE REVIEW

**Review by ribaraka — APPROVED** [review]

File: pkg/apiextensions/doc.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package apiextensions provides helpers structs which can be used anywhere
+// +kubebuilder:object:generate:=true
+package apiextensions
```
Comment (testisnullus) [inline pkg/apiextensions/doc.go:19]: Remove this file please

File: pkg/apiextensions/references.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	}
+}
+
+// LocalObjectReference is reference to an object in the known namespace
+type LocalObjectReference struct {
+	Name string `json:"name"`
+}
+
+// ObjectFieldReference is namespaced reference to the object field
+type ObjectFieldReference struct {
```
Comment (testisnullus) [inline pkg/apiextensions/references.go:40]: What does the ObjectField mean?

**Review by testisnullus — CHANGES_REQUESTED** [review]

File: pkg/apiextensions/doc.go (hunk 1 above)
Comment (worryg0d) [reply pkg/apiextensions/doc.go:19]: Removed

File: pkg/apiextensions/references.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+func (o *ObjectReference) AsNamespacedName() types.NamespacedName {
+	return types.NamespacedName{
+		Name:      o.Name,
+		Namespace: o.Namespace,
+	}
+}
```
[the remark this answers is not in the document]
Comment (worryg0d) [reply pkg/apiextensions/references.go:34]: We cannot use it because its fields doesn't have json tags

[the remark this answers is not in the document]
Comment (DoodgeMatvey) [reply pkg/apiextensions/references.go:34]: agree

**Review by DoodgeMatvey — APPROVED** [review]

**Review by testisnullus — APPROVED** [review]

**Pull request merged**